toolkit/tools/internal/pkggraph/pkggraph.go

```diff
@@ -1120,6 +1121,15 @@ func (g *PkgGraph) CreateSubGraph(rootNode *PkgNode) (subGraph *PkgGraph, err er
 	return
 }
 
+// IsSRPMAvailablePMC checks if an SRPM is prebuilt, returning true if so along with a slice of corresponding prebuilt RPMs.
+// The function will lock 'graphMutex' before performing the check if the mutex is not nil.
+func IsSRPMAvailablePMC(srpmPath string, pkgGraph *PkgGraph, graphMutex *sync.RWMutex, outDir, tmpDir, workerTar, existingRpmsDir, existingToolchainRpmsDir string, usePreviewRepo bool, repoFiles []string, disableUpstreamRepos bool, tlsClientKey string, tlsClientCert string) (isPrebuilt bool, expectedFiles, missingFiles []string) {
```

Comment:
>IsSRPMAvailablePMC [](http://example.com/codeflow?start=5&length=18)

It looks like `IsSRPMAvailablePMC` is only used inside this file. If that's the case, I'd keep make it an internal function instead of a public one. Especially since it looks like it needs an insane number of input arguments. We can always make it a public one, if we ever need it.